build-matrix.sh

```diff
@@ -11,57 +11,48 @@ HABANA_VERSIONS=("1.6.0-439")
-
-            # If the command failed and arch is x86_64, the z-stream doesn't exist and we can stop the loop.
-            # If the command failed and arch is not x86_64, we skip the image lookup.
-            if [ $? != 0 ]; then
-        	echo " Not found."
-        	if [ "${a}" == "x86_64" ]; then
-        	    break 2
-        	else
-        	    continue
-        	fi
-            fi
-            echo " Found."
+            echo -n "Get the release image for OCP ${z}-${a}."
+            IMG=$(oc adm release info --image-for=machine-os-content quay.io/openshift-release-dev/ocp-release:${z}-${a} 2>/dev/null)
 
     	    # Get the image info in JSON format, so we can use jq on it
             IMG_INFO=$(oc image info -o json -a ${PULL_SECRET_FILE} ${IMG} 2>/dev/null)
 
             # If the command failed, we skip the kernel lookup.
-            [ $? != 0 ] && echo "Image info for OCP 4.${y}.${z}-${a} not available" && continue
+            [ $? != 0 ] && echo "Image info for OCP ${z}-${a} not available" && continue
 
             # Add the kernel version from the image labels to the list of kernels
-            KVER=( $(echo ${IMG_INFO} | jq -r ".config.config.Labels[\"com.coreos.rpm.kernel\"]") )
+            KVER=$(echo ${IMG_INFO} | jq -r ".config.config.Labels[\"com.coreos.rpm.kernel-rt-core\"]" | sed -r "s/^(.*)\.rt.*(el[0-9]+.*).${a}/\1.\2/")
+            if [[ ! ${KVERS_OCPVERS_MAP[${KVER}]} ]]; then
+                KVERS_OCPVERS_MAP[${KVER}]=${z}
+            fi
             KVERS+=( ${KVER} )
-            echo "Kernel version for OCP 4.${y}.${z}-${a} is ${KVER}."
+            echo " Kernel version for OCP ${z}-${a} is ${KVER}.${a}."
         done
     done
 done
 
 # Remove duplicates from the list of kernels and sort it
 IFS=" " read -r -a KVERS <<< "$(tr ' ' '\n' <<< "${KVERS[@]}" | sort -u | tr '\n' ' ')"
 
-# Generate a list of unique kernels without arch
-IFS=" " read -r -a KVERS_NOARCH <<< "$(tr ' ' '\n' <<< "${KVERS[@]}" | sed "s/\..[^.]*$//" | sort -u | tr '\n' ' ')"
-
 # Initialize the matrix file
-echo -n "{ \"versions\": [" > ${MATRIX_FILE}
+echo "{" > ${MATRIX_FILE}
+echo "    \"versions\": [" >> ${MATRIX_FILE}
 
 # Build the matrix from the list of kernel versions
-LAST_KVER_NOARCH=""
+LAST_KVER=""
 COUNT=0
-for KVER_NOARCH in ${KVERS_NOARCH[@]}; do
+for KVER in ${KVERS[@]}; do
     # Extract RHEL version from the kernel version
-    RHEL_VERSION=$(echo ${KVER_NOARCH} | rev | cut -d "." -f 1 | rev | sed -e "s/^el//" -e "s/_/./")
+    RHEL_VERSION=$(echo ${KVER} | rev | cut -d "." -f 1 | rev | sed -e "s/^el//" -e "s/_/./")
 
     # Retrieve UBI image digest for RHEL version
-    UBI_DIGEST=$(oc image info -o json --filter-by-os "linux/amd64" registry.access.redhat.com/ubi8/ubi:${RHEL_VERSION} | jq .digest)
+    UBI_DIGEST=$(oc image info -o json --filter-by-os "linux/amd64" registry.access.redhat.com/ubi8/ubi-minimal:${RHEL_VERSION} 2>/dev/null | jq -r '.digest')
```

Comment:
You don't have `subscription-manager` in `ubi-minimal`. I'm not sure it will work.